src/services/expenseService.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable function-paren-newline */
+
+const expenses = [];
+
+const expenseServise = {};
+
+expenseServise.reset = () => {
+  expenses.length = 0;
+};
+
+expenseServise.getAll = (userId, categories, from, to) => {
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(
+      (exp) => exp.userId === Number(userId),
+    );
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter((exp) =>
+      categories.includes(exp.category),
+    );
+  }
+
+  if (from) {
+    const fromDate = new Date(from);
+
+    filteredExpenses = filteredExpenses.filter(
+      (exp) => new Date(exp.spentAt) >= fromDate,
+    );
+  }
+
+  if (to) {
+    const toDate = new Date(to);
+
+    filteredExpenses = filteredExpenses.filter(
+      (exp) => new Date(exp.spentAt) <= toDate,
+    );
+  }
+
+  return filteredExpenses;
+};
+
+expenseServise.getById = (id) => {
+  return expenses.find((outlay) => outlay.id === Number(id)) || null;
+};
+
+expenseServise.create = (body) => {
+  const { userId, title, amount, category, note, spentAt } = body;
+
+  const expense = {
+    id: Math.floor(Math.random() * 10000),
```

Comment:
Create function for id generation. This logic is not really good, because `id` can be repeated